server/jetstream_cluster_4_test.go

```diff
@@ -2549,3 +2549,359 @@ func TestJetStreamClusterMetaSyncOrphanCleanup(t *testing.T) {
 		require_Equal(t, state.Msgs, 10)
 	}
 }
+
+func TestJetStreamClusterConsumeWithStartSequence(t *testing.T) {
+
+	const (
+		NumMessages         = 10
+		ChosenSeq           = 5
+		StreamName          = "TEST"
+		StreamSubject       = "ORDERS.*"
+		StreamSubjectPrefix = "ORDERS."
+	)
+
+	for _, ClusterSize := range []int{
+		1, // Single server
+		3, // 3-node cluster
+	} {
+		R := ClusterSize
+		t.Run(
+			fmt.Sprintf("Nodes:%d,Replicas:%d", ClusterSize, R),
+			func(t *testing.T) {
+				// This is the success condition for all sub-tests below
+				var ExpectedMsgId = ""
+				checkMessage := func(t *testing.T, msg *nats.Msg) {
+					msgMeta, err := msg.Metadata()
+					require_NoError(t, err)
+
+					// Check sequence number
+					require_Equal(t, msgMeta.Sequence.Stream, ChosenSeq)
+
+					// Check message id
+					require_NotEqual(t, ExpectedMsgId, "")
+					require_Equal(t, msg.Header.Get(nats.MsgIdHdr), ExpectedMsgId)
+				}
+
+				checkRawMessage := func(t *testing.T, msg *nats.RawStreamMsg) {
+					// Check sequence number
+					require_Equal(t, msg.Sequence, ChosenSeq)
+
+					// Check message id
+					require_NotEqual(t, ExpectedMsgId, "")
+					require_Equal(t, msg.Header.Get(nats.MsgIdHdr), ExpectedMsgId)
+				}
+
+				// Setup: start server or cluster, connect client
+				var server *Server
+				if ClusterSize == 1 {
+					server = RunBasicJetStreamServer(t)
+					defer server.Shutdown()
+				} else {
+					c := createJetStreamCluster(t, jsClusterTempl, "HUB", _EMPTY_, ClusterSize, 22020, true)
+					defer c.shutdown()
+					server = c.randomServer()
+				}
+
+				// Setup: connect
+				var nc *nats.Conn
+				var js nats.JetStreamContext
+				nc, js = jsClientConnect(t, server)
+				defer nc.Close()
+
+				// Setup: create stream
+				_, err := js.AddStream(&nats.StreamConfig{
+					Replicas: R,
+					Name:     StreamName,
+					Subjects: []string{StreamSubject},
+				})
+				require_NoError(t, err)
+
+				// Setup: create subscriptions before stream is populated
+				var preCreatedSub, preCreatedSubDurable *nats.Subscription
+				{
+					preCreatedSub, err = js.PullSubscribe(
+						StreamSubject,
+						"",
+						nats.StartSequence(ChosenSeq),
+					)
+					require_NoError(t, err)
+					defer func() {
+						err := preCreatedSub.Unsubscribe()
+						if err != nil {
+							t.Error("Error unsubscribing", err)
```

Comment:
These can all just be `require_NoError(t, err)`, or if you want a one-liner, `require_NoError(t, preCreatedSub.Unsubscribe())` for example.